lib/game.rb

```diff
@@ -0,0 +1,89 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    #horizontal combos
+    [0,1,2], 
+    [3,4,5],
+    [6,7,8], 
+    #verical combos
+    [0,3,6], 
+    [1,4,7], 
+    [2,5,8],
+    #diagonal combos
+    [2,4,6],
+    [0,4,8]
+]
+  
+  def initialize(player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+    @player_1 = player_1 
+    @player_2 = player_2 
+    @board = board
+  end
+
+  def current_player
+    board.turn_count % 2 == 0 ? @player_1 : @player_2
+  end
+
+  def over?
+    draw? || won? 
+  end
+
+  def won?
+    WIN_COMBINATIONS.find do |combo|
+      @board.cells[combo[0]] == @board.cells[combo[1]] && 
+      @board.cells[combo[0]] == @board.cells[combo[2]] && 
+      @board.taken?(combo[0]+1)
+    end
+  end
+
+  def draw?
+    @board.full? && !won?
+  end
+
+  def winner
+    won? ? @board.cells[won?[0]] : nil
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.